qiskit_machine_learning/algorithms/classifiers/pegasos_qsvc.py

```diff
@@ -0,0 +1,285 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Pegasos Quantum Support Vector Classifier."""
+
+import logging
+from datetime import datetime
+from typing import Optional, Dict, Tuple
+
+import numpy as np
+from sklearn.svm import SVC
+
+from qiskit_machine_learning.exceptions import QiskitMachineLearningError
+from qiskit_machine_learning.kernels.quantum_kernel import QuantumKernel
+
+logger = logging.getLogger(__name__)
+
+
+class PegasosQSVC(SVC):
+    """
+    This class implements Pegasos Quantum Support Vector Classifier algorithm developed in [1]
+    and includes overridden methods ``fit`` and ``predict`` from the ``SVC`` super-class.
+
+    **Example**
+
+    .. code-block::
+        quantum_kernel = QuantumKernel()
+
+        pegasos_qsvc = PegasosQSVC(quantum_kernel=quantum_kernel)
+        pegasos_qsvc.fit(sample_train, label_train)
+        pegasos_qsvc.predict(sample_test)
+
+    **References**
+        [1]: Shalev-Shwartz et al., Pegasos: Primal Estimated sub-GrAdient SOlver for SVM.
+            `Pegasos for SVM <https://home.ttic.edu/~nati/Publications/PegasosMPB.pdf>`_
+
+    """
+
+    def __init__(
+        self,
+        quantum_kernel: Optional[QuantumKernel] = None,
+        C: float = 1,
+        num_steps: int = 1000,
+        precomputed: bool = False,
+    ) -> None:
+        """
+        Args:
+            quantum_kernel: QuantumKernel to be used for classification. Has to be ``None`` when
+                a precomputed kernel is used.
```

Comment:
I believe the SVC has various predefined strings and a callable for its kernel. Would it make sense to somehow combine the precomputed parameter and this to avoid having None here. E.g Optional[Union[QuantumKernel, str]]  - If None is the default one that is created, if "precomputed" then we do that, otherwise its a quantum kernel instance. Any other things are invalid. Maybe thats cleaner that having the interaction between two parameters?

Reply:
> 
> I believe the SVC has various predefined strings and a callable for its kernel. Would it make sense to somehow combine the precomputed parameter and this to avoid having None here. E.g Optional[Union[QuantumKernel, str]] - If None is the default one that is created, if "precomputed" then we do that, otherwise its a quantum kernel instance. Any other things are invalid. Maybe thats cleaner that having the interaction between two parameters?

Personally, I don't like type mixtures like `Union[QuantumKernel, str]`. We can mimic here the classical `SVC` and if you think it is better, I can do this.

Reply:
I guess it was partly a reaction to the wording on quantum_kernel specifically "Has to be None when...." but since that's the default its not as bad I guess as it sounds. I assume we want the quantum kernel param here to be just like the QSVC where None means we spin one up internally hence needing the extra flag (precomputed) to negate that. The one thing that is not so great about the suggestion I made is that it needs this magic string "precomputed" to be given - versus the boolean , but it did resemble the kernel parameter of SVC which is some union of string, callable. I don't have a strong opinion one way or the other - guess I was mainly trying to have something that was easier where it was either/or rather than don't do this if you do that.

Reply:
I'd keep as it it now if there's no a string objection. May be as an argument to keep it is that in `QSVC/QSVR` we don't accept a string as `quantum_kernel`, but only a kernel instance or ``None``.